libhirte/src/service/shutdown.c

```diff
@@ -0,0 +1,41 @@
+#include <stdio.h>
+#include <systemd/sd-bus-vtable.h>
+
+#include "../../include/common/common.h"
+#include "../../include/service/shutdown.h"
+#include "../common/memory.h"
+#include "service.h"
+
+int shutdown_event_loop(sd_event *event_loop) {
+        if (event_loop == NULL) {
+                return 0;
+        }
+
+        return sd_event_exit(event_loop, 0);
+}
+
+static int method_shutdown(sd_bus_message *m, void *userdata, UNUSED sd_bus_error *ret_error) {
```

Comment:
Do we want to have a file per service? 
Not sure if there is a proper C coding convention for this, but I don't see much files with this little functions in it.
On the other hand it makes navigation easier.

Reply:
Currently, I'd say yes. Having 2000+ lines of code in a single file is rather difficult to read and I'd reevaluate if we want to adhere to that convention. 
The idea behind the "service" is to split according to concerns:
- a function implementing the "business functionality" 
- and another function for registering it to dbus

By doing so, we can reuse the business function more easily. For example, in https://github.com/containers/hirte/pull/71 the signal handler could just import can call `shutdown`. A simplistic example, I know.

Reply:
On the other hand, having too many files for that little code also makes things a bit difficult

Reply:
Exactly. Lets stick with this first, and see how it evolves. 
Thinking about it, maybe we can move a lot of code from #71 here.

Reply:
Ack